File: app-java-fx/src/main/java/chessgame/app/fx/view/PieceView.java
package chessgame.app.fx.view;

import chessgame.facade.PieceFacade;
import javafx.geometry.Pos;
import javafx.scene.control.Label;
import javafx.scene.text.Font;

public final class PieceView extends Label {

    private final PieceFacade facade;

    private PieceView(PieceFacade facade) {
        super();
        this.facade = facade;
    }

    public static PieceView create(PieceFacade facade) {
        PieceView view = new PieceView(facade);

        view.setNeedsLayout(Boolean.TRUE);
        view.setLayoutX(1.0);
        view.setLayoutY(2.0);
        view.setPrefWidth(60.0);
        view.setPrefHeight(60.0);
        view.setAlignment(Pos.CENTER);
        view.setFont(Font.font(48.0));
        view.setMnemonicParsing(Boolean.TRUE);
        view.setWrapText(Boolean.TRUE);
        view.setText(facade.getSymbol());

        return view;
    }

    public PieceFacade getFacade() {
        return facade;
    }
}

File: app-kotlin-fx/src/main/kotlin/chessgame/app/fx/view/PieceView.kt
package chessgame.app.fx.view

import javafx.geometry.Pos.CENTER
import javafx.scene.control.Label
import javafx.scene.text.Font.font

class PieceView: Label() {
    companion object {
        fun of(symbol: String): PieceView {
            val view = PieceView()

            with (view) {
                isNeedsLayout = true
                layoutX = 1.0
                layoutY = 2.0
                prefWidth = 60.0
                prefHeight = 60.0
                alignment = CENTER
                font = font(48.0)
                isMnemonicParsing = true
                isWrapText = true
                text = symbol
            }

            return view
        }
    }
}